lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1082,9 +1082,12 @@ void FIRRTLLowering::run() {
   for (auto &op : body->getOperations()) {
     builder.setInsertionPoint(&op);
     builder.setLoc(op.getLoc());
-    if (succeeded(dispatchVisitor(&op))) {
+    auto done = succeeded(dispatchVisitor(&op));
+    if (!AnnotationSet(&op).empty())
```

Comment:
Does lowering remove annotations ever?  Can you do the check before the dispatchVisitor call?

Reply:
Lowering relies on and removed the `DontTouch` annotations. Just merged this change, https://github.com/llvm/circt/pull/1319

Reply:
Ok, makes sense!